test/unit/view-layout/new-test-layout.cpp

```diff
@@ -0,0 +1,412 @@
+  // Zero for J size should correctly produce projective layout
+  const my_layout layout(7, 0);
+
+  ASSERT_EQ(0, layout(0, 0));
+
+  ASSERT_EQ(1, layout(1, 0));
+  ASSERT_EQ(3, layout(3, 0));
+
+  // J should be projected out
+  ASSERT_EQ(0, layout(0, 1));
+  ASSERT_EQ(0, layout(0, 5));
+
+  TypeParam pK = 0;
+  // Check that we get the identity (mod 7)
+  for (int k = 0; k < 20; ++k) {
+
+    // inverse map
+    TypeParam i, j;
+    layout.toIndices(pK, i, j);
+
+    // forward map
+    TypeParam k2 = layout(i, j);
+
+    // check ident
+    ASSERT_EQ(pK % 7, k2);
+
+    // check projection of j
+    ASSERT_EQ(j, 0);
+    pK++;
+  }
+}
+
+TYPED_TEST(LayoutTest, 3D_KJI_ProjJ)
+{
+  //using my_layout = RAJA::Layout<3>;
+  using my_layout = RAJA::TypedLayout<TypeParam, RAJA::tuple<TypeParam, TypeParam, TypeParam>>;
+
+  /*
+   * Construct a 3D projective layout:
+   *
+   * I is stride 1
+   * J is stride 0  -  projected out
+   * K is stride 3
+   *
+   * Linear indices range from [0, 21)
+   *
+   * values of J should have no effect on linear index
+   *
+   * All linear indices should produce J=0
+   *
+   */
+
+  // Construct using variadic "sizes" ctor
+  // Zero for J size should correctly produce projective layout
+  const my_layout layout =
+      RAJA::make_permuted_layout({{3, 0, 7}},
+                                 RAJA::as_array<RAJA::PERM_KJI>::get());
+
+  ASSERT_EQ(0, layout(0, 0, 0));
+
+  ASSERT_EQ(1, layout(1, 0, 0));
+  ASSERT_EQ(3, layout(3, 0, 0));
+
+  // J should be projected out
+  ASSERT_EQ(0, layout(0, 1, 0));
+  ASSERT_EQ(0, layout(0, 5, 0));
+
+  ASSERT_EQ(6, layout(0, 0, 2));
+  ASSERT_EQ(12, layout(0, 0, 4));
+
+  // Check that we get the identity (mod 21)
+  for (TypeParam x = 0; x < 40; ++x) {
+
+    // inverse map
+    TypeParam i, j, k;
+    layout.toIndices(x, i, j, k);
+
+    // forward map
+    TypeParam x2 = layout(i, j, k);
+
+    // check ident
+    ASSERT_EQ(x % 21, x2);
+
+    // check projection of j
+    ASSERT_EQ(j, 0);
+  }
+}
+
+TEST(LayoutTest, 2D_StrideOne)
+{
+  using my_layout = RAJA::Layout<2>;
+  using my_layout_s1 = RAJA::Layout<2, ptrdiff_t, 0>; // first index is stride-1
+
+  /*
+   * Construct a 2D layout:
+   *
+   * I is stride 1
+   * J is stride 3
+   *
+   * Linear indices range from [0, 15)
+   *
+   */
+  const my_layout layout =
+      RAJA::make_permuted_layout({{3, 5}},
+                                 RAJA::as_array<RAJA::PERM_JI>::get());
+
+
+  /*
+   * Construct another 2D layout that forces J to be stride-1
+   */
+  const my_layout_s1 layout_s1 = layout;
+
+
+  // Check that we get the same layout
+  for (int i = 0; i < 3; ++i) {
+    for (int j = 0; j < 15; ++j) {
+
+      ASSERT_EQ(layout(i, j), layout_s1(i, j));
+    }
+  }
+}
+
+
+TEST(LayoutTest, 2D_StaticLayout)
+{
+  RAJA::Layout<2> dynamic_layout(7, 5);
+  using static_layout = RAJA::TypedStaticLayout<RAJA::PERM_IJ,RAJA::tuple<int,int>,7,5>;
+  
+  //Breaks here...
+  static_layout::s_oper(0,0);
```

Comment:
@MrBurmark @ajkunen  , this won't compile I was wondering if you might have some insight on the issue? 

I hit the
 following error: 
```
 error: incomplete type 'static_layout {aka RAJA::detail::TypedStaticLayoutImpl<RAJA::detail::StaticLayoutBase_impl<camp::int_seq<long int, 0l, 1l>, camp::int_seq<long int, 7l, 5l>, camp::int_seq<long int, 5l, 1l> >, camp::tuple<int, int> >}' used in nested name specifier
```